src/lib/storage/lz4_segment/lz4_encoder.hpp

```diff
@@ -168,37 +165,40 @@ class LZ4Encoder : public SegmentEncoder<LZ4Encoder> {
      * The offsets are stored as 32 bit unsigned integer as opposed to 64 bit (size_t) so that they can later be
      * compressed via vector compression.
      */
-    auto offsets = pmr_vector<uint32_t>{alloc};
-    offsets.resize(num_elements);
+    auto offsets = pmr_vector<uint32_t>{allocator};
 
     /**
      * These are the lengths of each string. They are needed to train the zstd dictionary.
      */
-    auto sample_sizes = pmr_vector<size_t>{alloc};
-    sample_sizes.resize(num_elements);
+    auto string_samples_lengths = pmr_vector<size_t>{allocator};
```

Comment:
I probably asked this when we originally brought this is, but how are these “samples” if we put in every single value?

Reply:
The values are(/might be) later "sampled" (taking the `n` first values) within `_train_dictionary(const pmr_vector<T>& values, const pmr_vector<size_t>& sample_sizes)`.

Reply:
There are several issues with the current LZ4 segment implementation, which I plan to rewrite. Shall I open an issue with an overview or do you trust me to address these issues in an upcoming PR?

Amongst these things are the removal of offsets, larger blocks (we basically never fill up the 4 KB), "real" sampling (tracking unique values and using the most often occuring values when samples too large), and more.